samtranslator/model/api/api_generator.py

```diff
@@ -593,8 +593,6 @@ def _construct_usage_plan(self, rest_api_stage=None):
         # create a usage plan for all the Apis
         elif create_usage_plan == "SHARED":
             usage_plan_logical_id = "ServerlessUsagePlan"
-            if self.logical_id not in ApiGenerator.depends_on_shared:
-                ApiGenerator.depends_on_shared.append(self.logical_id)
```

Comment:
I believe the original definition of `ApiGenerator.depends_on_shared` can be removed, as well. It's not used anywhere else.